### DISCUSSION TITLE
Anvil extras Slider, connect property not updating in UI

### DISCUSSION THREAD

**Discussion by morod3v** [anvilistas/anvil-extras#537] [Q&A]
Hello dear community

**What I'm trying to do:**

I am trying to modify the anvil extras' slider component "connect" property from code.

**What I've tried and what's not working:**

I have a slider with two handles.
Initially, the connect property is set to True, such that the inner connection is colored.
In the code, i've tried this:
```python
self.slider_1.connect = [True, False, True]
print(self.slider_1.connect)
```
which does work (i got [True, False, True] printed)

HOWEVER
In the UI, the color of the slider doesn't change, and i don't get the desired connection (colored, not colored, colored)

It does work when i do it in the designer.
But i really need it to work from code because i need it to be triggered from an other component click event.

**Other properties:**

Here are some screenshots of the other properties of the slider.

![Screenshot 2024-06-08 173054](https://github.com/anvilistas/anvil-extras/assets/157087595/4ebdbba9-b4c8-43bb-8c5a-936a8cee0425)
![Screenshot 2024-06-08 173112](https://github.com/anvilistas/anvil-extras/assets/157087595/a5b6e4cd-ce1e-490d-96e0-641314614e98)

Any idea of where this could come from ?

**Comment by s-cork** [comment]
following from anvil forum

Yes we never implemented connect as a dynamic property, because it's not a dynamic property in the underlying javascript library
We'll need to destroy the existing slider component and create a new one whenever the connect property changes
(like we do in the designer)

**Reply by s-cork** [reply]
@morod3v - were you interested in writing a PR? If so let us know and I can point you in the right direction. If not - no problem. I'll take a look at this when i get some time next week hopefully,

**Reply by morod3v** [reply]
Yup, i've saw your answer in the anvil forum
And you've made me try exactly what you said : destroy the old slider and create a new one, with the same properties as the old one, except for the connect property which i change as desired.
Good workaround.

PS: as i stated in my bug report, ... what's a PR ???

**Reply by s-cork** [reply]
It's a pull request. A piece of code that you would contribute to fix the issue.

**Reply by morod3v** [reply]
Hey @s-cork !

Sorry for late answer

Sure, I'll be glad to contribute if I can !
Just let me know
